policy/policy.go

```diff
@@ -44,85 +45,198 @@ func (policy *PolicySvc) Routes() common.Routes {
+	if endpoint.TenantID != 0 {
+		tenantIDToUse = string(endpoint.TenantID)
+	} else if endpoint.TenantExternalID != "" {
+		tenantIDToUse = endpoint.TenantExternalID
+	}
+	if tenantIDToUse != "" {
+		tenantsUrl := fmt.Sprintf("%s/tenants/%s", tenantSvcUrl, tenantIDToUse)
+		err = policy.client.Get(tenantsUrl, &ten)
+		if err != nil {
+			return err
+		}
+		endpoint.TenantNetworkID = ten.Seq
+	}
+
+	var segmentIDToUse string
+	if endpoint.SegmentID != 0 {
+		segmentIDToUse = string(endpoint.SegmentID)
+	} else if endpoint.SegmentExternalID != "" {
+		segmentIDToUse = endpoint.SegmentExternalID
 	}
-	// Get host info from topology service
-	topoUrl, err := client.GetServiceUrl("topology")
+	if segmentIDToUse != "" {
+		tenantsUrl := fmt.Sprintf("%s/tenants/%s/segment/%s", tenantSvcUrl, ten.ID, segmentIDToUse)
+		segment := &tenant.Segment{}
+		err = policy.client.Get(tenantsUrl, &segment)
+		if err != nil {
+			return err
+		}
+		endpoint.SegmentNetworkID = segment.Seq
+	}
+	return nil
+}
+
+// augmentPolicy augments the provided policy with information gotten from
+// various services.
+func (policy *PolicySvc) augmentPolicy(policyDoc *common.Policy) error {
+	// Get info from topology service
+	topoUrl, err := policy.client.GetServiceUrl("topology")
 	if err != nil {
-		return nil, err
+		return err
 	}
 
+	// Query topology for data center information
+	// TODO move this to root
 	index := common.IndexResponse{}
-	err = client.Get(topoUrl, &index)
+	err = policy.client.Get(topoUrl, &index)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	tenantSvcUrl, err := client.GetServiceUrl("tenant")
+	dcURL := index.Links.FindByRel("datacenter")
+	dc := common.Datacenter{}
+	err = policy.client.Get(dcURL, &dc)
 	if err != nil {
-		return nil, err
+		return err
+	}
+	log.Printf("Policy server received datacenter information from topology service: %v\n", dc)
+	policyDoc.Datacenter = dc
+
+	for _, endpoint := range policyDoc.AppliedTo {
+		err = policy.augmentEndpoint(&endpoint)
+		if err != nil {
+			return err
+		}
 	}
 
-//	for _, endpoint := range policyDoc.AppliedTo {
-//	}
-	tenantsUrl := fmt.Sprintf("%s/tenants", tenantSvcUrl)
-	var tenants []tenant.Tenant
-	err = client.Get(tenantsUrl, &tenants)
+	for _, endpoint := range policyDoc.Peers {
+		err = policy.augmentEndpoint(&endpoint)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// distributePolicy distributes policy to all agents.
+// TODO how should error handling work here really?
+func (policy *PolicySvc) distributePolicy(policyDoc *common.Policy) error {
+	hosts, err := policy.client.ListHosts()
 	if err != nil {
-		return nil, err
+		return err
 	}
-	found := false
-	var i int
-	tenantName := "TODO"
-	for i = range tenants {
-		if tenants[i].Name == tenantName {
-			found = true
-			break
+	errStr := make([]string, 0)
+	for _, host := range hosts {
+		// TODO make schema configurable
+		url := fmt.Sprintf("http://%s:%d/policies", host.Ip, host.AgentPort)
```

Comment:
This will try co contact Romana agent on a host while it should really be contacting policy agent. The are not the same thing yet :-(
Policy agent runs on 9630 by default, we can use this port or pick another one but distinct from romana agent ones.